backend/core/detection.py

```diff
@@ -819,6 +825,54 @@ def detect_path_type_1_hijack(
                 return (-1, "-")
             return (first_neighbor_asn, "1")
 
+        @exception_handler(log)
+        def detect_path_type_P_hijack(
+            self,
+            monitor_event: Dict,
+            prefix_node: Dict,
+            prefix_node_conf: Dict,
+            *args,
+            **kwargs
+        ) -> Tuple[int, str]:
+            """
+            Type-P hijack detection.
+            """
+            if "orig_path" not in monitor_event:
+                return (-1, "-")
+            orig_path = monitor_event["orig_path"]
+            pattern_matched = False
+            pattern_hijacker = -1
+            best_match_length = 0
+            if len(prefix_node_conf["prepend_seq"]):
+                for conf_seq in prefix_node_conf["prepend_seq"]:
+                    if len(orig_path) >= len(conf_seq) + 1:
+                        monitor_event_seq = orig_path[
+                            len(orig_path) - len(conf_seq) - 1 : -1
+                        ]
+                        if monitor_event_seq == conf_seq:
+                            pattern_matched = True
+                            break
+                        else:
```

Comment:
a comment here to make it clear what the else statement does so we know

Reply:
Done this, please check the misc comments in-code If the explanation is comprehensive, please resolve. Otherwise let me know to update the comment(s).